src/steps/fetch-users/index.ts

```diff
@@ -0,0 +1,218 @@
+import {
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  createIntegrationEntity,
+  createIntegrationRelationship,
+} from '@jupiterone/integration-sdk';
+import { HerokuClient } from '../../heroku';
+import {
+  HerokuTeamMember,
+  HerokuEnterpriseMember,
+  HerokuUser,
+} from '../../types/herokuTypes';
+import { getIdsFromJobState } from '../../util/getIdsFromJobState';
+
+// note: No /users endpoint exists to fetch all users.
+// WARNING: Heroku labels these APIs as in DEVELOPMENT
+// https://devcenter.heroku.com/articles/platform-api-reference#team-member
+// https://devcenter.heroku.com/articles/platform-api-reference#enterprise-account-member
+//
+// Heroku labels this API as in PRODUCTION
+// https://devcenter.heroku.com/articles/platform-api-reference#account-info-by-user
+const step: IntegrationStep = {
+  id: 'fetch-users',
```

Comment:
This step does a lot more than just fetch users. 😆 I think it's fine for now, but I imagine this could be broken into a few steps, each doing less work, and then another step that `dependsOn` some smaller steps to build relationships.

Reply:
Haha, very true. Since the heroku API doesn't expose users directly, I was trying to get all of them in one step to avoid duplication. But I agree it's too big and I'll break it down into a few different steps.